read-fonts/src/collections/int_set/bitset.rs

```diff
@@ -29,13 +30,27 @@ const PAGE_BITS_LOG_2: u32 = PAGE_BITS.ilog2();
 /// A fast, efficient, sparse, & ordered `u32` set.
 ///
 /// For a higher-level API that supports inversion and generic int types, use [`super::IntSet`]
-#[derive(Clone, Debug, Default)]
+#[derive(Debug, Default)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct U32Set {
     // TODO(garretrieger): consider a "small array" type instead of Vec.
     pages: Vec<BitPage>,
     page_map: Vec<PageInfo>,
     length: u64,
+    last_page_map_index: AtomicUsize,
```

Comment:
you have custom impls of Hash/PartialEq/PartialOrd etc? because this is what clippy is complaining about; because `IntSet` is used as a key in some hashmaps in various places, the fact that it now has interior mutability could in theory cause headaches, since the key could be mutated in such a way as to violate the invariants of the collection types.

Reply:
Ah makes sense, yeah I'll add custom impls of all of those that ignore the atomic.

Reply:
Updated to add the clippy allows as needed. Also flagged the interior mutable field not to be serialized.